extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/runtime/interceptor/TransactionalInterceptorBase.java

```diff
@@ -34,6 +40,8 @@
 public abstract class TransactionalInterceptorBase implements Serializable {
 
     private static final long serialVersionUID = 1L;
+    private static final Logger log = Logger.getLogger(TransactionalInterceptorBase.class);
+    private final Map<Method, Integer> cache = new ConcurrentHashMap<>();
```

Comment:
Maybe make it static

Reply:
I'm not sure using Method as cache key is a good idea here, the method object is not very lightweigth and we may retain it for a long time.

Reply:
My experience with HV is: caching things at a method level costs a lot. Because you either use `Method` or generate a full signature and in both cases, it's not fast.

As for making it static, depends. If you make it static, you will need to be able to clean it properly at shutdown.